### PR TITLE
MBUS: added support for programming via mbus

### PR DESCRIPTION
**Pull request by lukefahr**
Squashed commit of the following:

commit d01accf1d422c2dcc18052c9514982c1f476f001
Author: Andrew Lukefahr <lukefahr@umich.edu>
Date:   Wed Jun 21 12:27:58 2017 -0400

    MBUS:  refactor programmer code

commit 29f79b123a2596277510de0f40272e5728cec2db
Author: Andrew Lukefahr <lukefahr@umich.edu>
Date:   Tue Jun 20 17:41:42 2017 -0400

    ICE: mbus_programmer works!

commit 753d179b05fdc3d30d8a4c33aae9a49ea99322aa
Author: Andrew Lukefahr <lukefahr@umich.edu>
Date:   Tue Jun 20 17:40:48 2017 -0400

    ICE:  backed timeout to 500ms b/c of cygwin

commit 088a2ceb9a51f1be0f52c47694865386991dbde4
Merge: 8946efe f37d687
Author: Andrew Lukefahr <lukefahr@umich.edu>
Date:   Wed May 24 14:52:36 2017 -0400

    Merge branch 'master' into mbus_programmer

commit 8946efeac5780a888f37d7aa2f36501dce47a6c5
Merge: d4106d7 2987c5f
Author: Andrew Lukefahr <lukefahr@umich.edu>
Date:   Tue May 16 11:17:18 2017 -0400

    Merge branch 'master' into mbus_programmer

commit d4106d75a501980df32a02a91d14a12241efe156
Author: Andrew Lukefahr <lukefahr@umich.edu>
Date:   Thu May 11 16:23:58 2017 -0400

    Tests: added automated test for timeout

commit 8b593eed7a7acf0762b2b80ba87a4ff3bb8c25dc
Author: Andrew Lukefahr <lukefahr@umich.edu>
Date:   Thu May 11 13:25:13 2017 -0400

    ICE/ICE_SIM:  added support to for timeout/'e' message

commit 69c93cb44098a4c0263d641e1d5d0cbdd99c2b22
Author: Andrew Lukefahr <lukefahr@umich.edu>
Date:   Thu May 11 10:55:48 2017 -0400

    ICE:  added a timeout for recv

commit 04a7775be6efc4685c4c512136d1b444611c022f
Merge: 34e5731 88b9add
Author: Andrew Lukefahr <lukefahr@umich.edu>
Date:   Wed May 10 12:00:25 2017 -0400

    Merge branch 'master' into mbus_programmer

commit 34e5731e85332fc043a6e97b331577ea57b0b67d
Author: Andrew Lukefahr <lukefahr@umich.edu>
Date:   Tue May 9 16:45:48 2017 -0400

    DEV: broken commit

commit a381383c8ecbb2cde236334f0b57922fd9aa3d08
Author: Andrew Lukefahr <lukefahr@umich.edu>
Date:   Tue May 9 16:32:59 2017 -0400

    ICE: bump supported version to 0.4

commit ee6e77f27d519ccf441e355c1d3ab1d991434411
Author: Andrew Lukefahr <lukefahr@umich.edu>
Date:   Mon May 1 12:20:32 2017 -0400

    mbus_programmer: started working on actually sending data

commit 563e1b62a167794cca15a327a5ff19a8ca8474bb
Author: Andrew Lukefahr <lukefahr@umich.edu>
Date:   Thu Apr 27 15:24:35 2017 -0400

    mbus_programmer:  works with m3_ice_simulator

commit 612c3a5fa0606b0e35056839e6c567e92c300cfb
Merge: 1a82fba 0925717
Author: Andrew Lukefahr <lukefahr@umich.edu>
Date:   Thu Apr 27 13:11:51 2017 -0400

    Merge branch 'master' of https://github.com/mbus/m3-python into mbus_programmer

commit 1a82fba2c5061ea3aec2ffd17922792415ca0dd4
Author: Andrew Lukefahr <lukefahr@umich.edu>
Date:   Tue Apr 25 17:03:31 2017 -0400

    tmp work stuff

### CODE REVIEW

File: m3/ice.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 import struct
 import sys
 import time
+import os
```
Comment (ppannuto) [inline m3/ice.py]: nit: os already imported

File: m3/ice.py [hunk 2]
```diff
@@ -1432,6 +1434,13 @@ def mbus_set_master_onoff(self, onoff):
         DEFAULT: OFF
         '''
         self.min_version(0.2)
+        if isinstance(onoff, str):
+            onoff = True if onoff in \
+                    ['On', 'on'] else False
```
Comment (ppannuto) [inline m3/ice.py]: what about `'ON'` ? :D

File: m3/ice.py (hunk 1 above)
Comment (lukefahr) [inline m3/ice.py]: fixed, can you "squash and merge"?

**Comment by ppannuto** [comment]
image

Done -- you can too, it's the little down arrow here

**Pull request merged**